### ISSUE TITLE
Accessibility: Set TITLE for Navigation Block automatically with option to customize

### ISSUE THREAD

**Issue by jhmonroe** [WordPress/gutenberg#72443]
### Description

Piggy-backing on "Option to set title for the Table of Contents block" [#52378](https://github.com/WordPress/gutenberg/issues/52378) so hopefully both can be implemented together to improve accessiblity  

Someone please tag as Accessibility! I can't?

Aside from the TOC, the block editor doesn't seem to output a TITLE for any Navs. It uses the Navigation name "ie: Primary Menu" as the aria-label which is great.

See image below of what is shown in the HTML5 outline (ie: in inspector or when checking using the HTML5 Outliner browser extension.

<img width="450" height="412" alt="Image" src="https://github.com/user-attachments/assets/5049525a-53fb-4204-bc72-4ce4b496b93f" />

Here I have page title, Nav, then a series of Headings in the main content:

<img width="278" height="150" alt="Image" src="https://github.com/user-attachments/assets/9b780c81-d45c-4dd7-ab62-dc1c7c83eba7" />

Title attributes are able to be set in the right sidebar for individual items inside the Nav, but not for the overall Nav itself.

A simple solution for the overarching Nav would be to pull the Nav name as the Title attribute. The same way it does for aria-label.
Then the user could also customize it if needed.

After searching all a11y tickets for nav title I came across this interesting issue (https://github.com/WordPress/gutenberg/pull/68735) addressed by @t-hamano and others about how these things are pulled/saved/used automatically for aria-label. Perhaps this will inform the approach taken for the TITLE attribute for all Navs.

### Step-by-step reproduction instructions

- add a nav with custom name
- save and view in browser
- view the DOM in web browser inspector or HTML5 Outliner extension
- note that it says "Untitled NAV" even though the nav has a name in Wordpress and the aria-label is present

### Environment info

- Wordpress 6.8.3
- Site Editor

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by jorgefilipecosta** [comment]
I was able to replicate this, I think using the nav name as the title makes total sense. I will try to propose a solution.

**Comment by jorgefilipecosta** [comment]
Unfortunately after looking deeper title may not the right approach here, title provides a tooltip with the navigation menu name which may not be something we want to expose to users visually.
The navigation already has a an aria-label so it is being properly labeled. The HTML5 document outline algorithm looks for heading elements so I think if we want to fix the "Untittled Nav" on the outline issue the correct fix would be something like:
```
<nav aria-label="Primary Menu">
  <h2 class="screen-reader-text">Primary Menu</h2>
  <!-- nav content -->
</nav>
````
Or 
```
<nav aria-labelledby="nav-heading-123">
  <h2 id="nav-heading-123" class="screen-reader-text">Primary Menu</h2>
  <!-- nav content -->
</nav>
```

But then it raises the question of what would be the correct heading level to use, I guess it depends on the situation, cc: @WordPress/accessibility in case the team has some insights on what paths we have to fix this issue.

**Comment by joedolson** [comment]
I'm not sure exactly what the original request is for. The `title` attribute is a largely useless attribute, used as a 3rd-tier option for accessible name if there is no other information available, but unavailable entirely in most situations.

What is the screenshot that shows the "Untitled Nav" coming from? Your nav is titled, using the `aria-label`. The alternate proposal by @jorgefilipecosta, using `aria-labelledby` and a heading, is a common pattern - Drupal uses this a lot. It has the advantage that the navigation menu heading will appear in the heading hierarchy, but the downsides that 1) in many AT/browser combinations, the name will be announced twice (once on entering the nav, and once reading the heading), and 2) you have to manage the heading level, as not all nav element are in the same position in the hierarchy.

The notes say "view the DOM in web browser inspector or HTML5 Outliner extension"; in the DOM, I see a navigation with Primary menu on the site in question. I looked at the HTML 5 Outliner extension, and I'm dubious about how useful that is. It hasn't been updated meaningfully in 5 years, and it's explicitly articulating the HTML5 outline algorithm, which was a proposed feature for HTML5 that was never implemented in any browser or assistive technology, and is not useful for testing. 

In my opinion, this issue should be closed.

**Comment by jhmonroe** [comment]
> I'm not sure exactly what the original request is for. The `title` attribute is a largely useless attribute, used as a 3rd-tier option for accessible name if there is no other information available, but unavailable entirely in most situations.
> 
> What is the screenshot that shows the "Untitled Nav" coming from? Your nav is titled, using the `aria-label`. The alternate proposal by [@jorgefilipecosta](https://github.com/jorgefilipecosta), using `aria-labelledby` and a heading, is a common pattern - Drupal uses this a lot. It has the advantage that the navigation menu heading will appear in the heading hierarchy, but the downsides that 1) in many AT/browser combinations, the name will be announced twice (once on entering the nav, and once reading the heading), and 2) you have to manage the heading level, as not all nav element are in the same position in the hierarchy.
> 
> The notes say "view the DOM in web browser inspector or HTML5 Outliner extension"; in the DOM, I see a navigation with Primary menu on the site in question. I looked at the HTML 5 Outliner extension, and I'm dubious about how useful that is. It hasn't been updated meaningfully in 5 years, and it's explicitly articulating the HTML5 outline algorithm, which was a proposed feature for HTML5 that was never implemented in any browser or assistive technology, and is not useful for testing.
> 
> In my opinion, this issue should be closed.

That context of HTML5 is helpful! I'm just using the accessibility checking tools that I learned about from @amberhinds and others from the WP Accessibility Day 2025. Perhaps she will have some thoughts about how relevant this still is?

**Comment by amberhinds** [comment]
As noted above the `aria-label` provides an accessible name for the navigation menu. Adding a `title` attribute would not be recommended, as it would likely result in duplicate announcements in some screen readers or an undesired tooltip in some browsers.

The screenshot above from the HTML5 Outliner browser extension is what I consider a false positive or a misunderstanding by the browser extension developer (if they're treating a missing title attribute as an error). We actively advise people not to use `title` attributes to name components and frequently recommend removing them.

In my opinion, this issue should be closed as invalid or "won't fix."

**Comment by jhmonroe** [comment]
Thank you!! If I see this extension recommended again in any livestreams etc I will notify everyone to take with a grain of salt! (I'm pretty sure it was recommended in one of the accessibility day 2025 videos since that's when I started diving deeper into all of this) Will also attempt to contact the developer.

> As noted above the `aria-label` provides an accessible name for the navigation menu. Adding a `title` attribute would not be recommended, as it would likely result in duplicate announcements in some screen readers or an undesired tooltip in some browsers.
> 
> The screenshot above from the HTML5 Outliner browser extension is what I consider a false positive or a misunderstanding by the browser extension developer (if they're treating a missing title attribute as an error). We actively advise people not to use `title` attributes to name components and frequently recommend removing them.
> 
> In my opinion, this issue should be closed as invalid or "won't fix."

**Comment by t-hamano** [comment]
The title attribute is considered deprecated and unnecessary, so let's close this issue.

**Comment by jorgefilipecosta** [comment]
Thank you all for the insights!
